### PR TITLE
Flux – query data

### LINKED ISSUE #2621: Missing basic concepts in Flux docs?
I'm transitioning from InfluxQL to Flux and am having a harder time than I think is warranted.

I've read the [Get started with Flux](https://docs.influxdata.com/influxdb/v2.0/query-data/get-started/), [Query InfluxDB with Flux](https://docs.influxdata.com/influxdb/v2.0/query-data/get-started/query-influxdb/), [Query data with Flux](https://docs.influxdata.com/influxdb/v2.0/query-data/flux/) pages several times, as well as [Querying fields and tags](https://docs.influxdata.com/influxdb/v2.0/query-data/flux/query-fields/).

I haven't seen it mentioned anywhere explicitly,
* Why some record fields in the filter `fn` start with an underscore. Presumably `_measurement` is reserved, but why also `_field`? In InfluxDB, a record has a [field set](https://docs.influxdata.com/influxdb/v1.8/concepts/glossary/#field-set), so there should be multiple fields?
* What all these `_`-prefixed record fields are
* How to query for a record with a given field value (I know that's not a recommended practice in InfluxQL, but it's possible)

I also miss the examples from v1, and IMO the [key concepts explanation from v1.8](https://docs.influxdata.com/influxdb/v1.8/concepts/key_concepts/) was far clearer and better illustrated with the butterflies and water levels examples. The v2 Flux docs feel more arid, seem to assume the reader is familiar with a lot of concepts, and its [Key concepts](https://docs.influxdata.com/influxdb/v2.0/query-data/get-started/#key-concepts) section is tiny.

### PR DESCRIPTION
**Pull request by sanderson**
Closes #2621 

Adds a new section in the Flux docs for querying different data sources. The first data source included with this PR is InfluxDB.

- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

File: content/flux/v0.x/query-data/influxdb.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+To query InfluxDB 1.x, use the `database-name/retention-policy-name` naming
+convention for your bucket name.
+For example, to query data from the `autogen` retention policy in the `telegraf` database:
+
+```js
+from(bucket: "telegraf/autogen")
+  |> range(start: -30m)
+```
+
+To query the **default retention policy** in a database, use the same bucket naming
```
Comment (kelseiv) [inline content/flux/v0.x/query-data/influxdb.md:63]: Shall we provide context for default retention policy? 
/influxdb/v1.8/query_language/explore-schema/#run-a-show-retention-policies-query-with-the-on-clause
or here?: 
/influxdb/v1.8/query_language/manage-database/#create-a-retention-policy

File: content/flux/v0.x/query-data/influxdb.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+convention, but do not provide a retention policy:
+
+```js
+from(bucket: "telegraf/")
+  |> range(start: -30m)
+```
+
+
+## Results structure
```
Comment (kelseiv) [inline content/flux/v0.x/query-data/influxdb.md:72]:
```suggestion
### Results structure
```

File: content/flux/v0.x/query-data/influxdb.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+- **_measurement**: Measurement name
+- **_field**: Field key
+- **_value**: Field value
+- **Tag columns**: A column for each tag where the column label is the tag key
+  and the column value is the tag value
+
+{{% note %}}
+#### Columns with the underscore prefix
+Columns with the underscore (`_`) prefix are considered "system" columns.
+Some Flux functions require these columns to function properly.
```
Comment (kelseiv) [inline content/flux/v0.x/query-data/influxdb.md:90]:
```suggestion
Some Flux functions require these columns.
```

**Review by kelseiv — APPROVED** [review]

File: content/flux/v0.x/query-data/influxdb.md (hunk 1 above)
Comment (sanderson) [inline content/flux/v0.x/query-data/influxdb.md:63]: Yeah, definitely.

```suggestion
To query the [default retention policy](/{{< latest "influxdb" "v1" >}}/query_language/manage-database/#create-a-retention-policy) in a database, use the same bucket naming
```

**Pull request merged**